hermeto/core/package_managers/maven/main.py

```diff
@@ -331,12 +331,47 @@ def _get_maven_dependencies(
             "repository_id": repository_id,
             "url": url
         })
+        
+        # WORKAROUND: Also track .pom files for _remote.repositories
+        pom_filename = Path(filename).stem + ".pom"
```

Comment:
Two quick observations about this function:

1. It is already almost 150 lines long and keeps growing and
2. There are multiple one-line comments explaining what the next code of block is about to do.

The standard rule for 2. is to split out a helper each time a comment explaining what a block of code is about to do appears. This would also help with 1.

Reply:
@a-ovchinnikov I think at this point there already are quite a few substantial comments and this looks like still very much like a live WIP far from a regular proposal, so I think you don't need to spend your precious time digging too deep into function-level stuff right now :)